### PR TITLE
plat-k3: am62x: add SA2UL and TRNG support

### PR DESCRIPTION
**Pull request by ti-kamlesh**
Add SA2UL and TRNG support for TI SoC AM62X through OP-TEE.

Signed-off-by: Kamlesh Gurudasani <kamlesh@ti.com>

### CODE REVIEW

**Comment by jforissier** [comment]
The subject says this adds AM62x support but it is not true since commit bf1abbb3cd01 has already introduced AM62x in CI. So I would prefer: `plat-k3: am62x: add SA2UL and TRNG support`. Or would AM62x build but not run properly with the current config? In that case this change would be a fix (`plat-k3: am62x: fix SA2UL and TRNG support`)?
It may not matter much but with the current description it is impossible to know what is wrong and what is added or fixed.

Other than that:
`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/plat-k3/platform_config.h [hunk 1]
```diff
@@ -63,26 +63,38 @@
...
 #elif defined(PLATFORM_FLAVOR_j784s4)
 #define SA2UL_BASE		0x40900000
 #define SA2UL_TI_SCI_DEV_ID	-1
 #define SA2UL_TI_SCI_FW_ID	1196
+#define SA2UL_TI_SCI_FW_RGN_ID	0
 #elif defined(PLATFORM_FLAVOR_am64x)
 #define SA2UL_BASE		0x40900000
 #define SA2UL_TI_SCI_DEV_ID	133
 #define SA2UL_TI_SCI_FW_ID	35
+#define SA2UL_TI_SCI_FW_RGN_ID	0
```
Comment (jforissier) [inline core/arch/arm/plat-k3/platform_config.h:81]: Sorry to comment on a commit that I have already acked... but those `#define SA2UL_TI_SCI_FW_RGN_ID 0` affect all non-am62x flavors. Are they legitimate here? Shouldn't they be part of a different commit? (what does this fix?)

Comment (ti-kamlesh) [reply core/arch/arm/plat-k3/platform_config.h:81]: Not an issue @jforissier, anything to improve the quality. Thanks for the comment.

#define SA2UL_TI_SCI_FW_RGN_ID 0 had same value for all flavors earlier
am62x has different value SA2UL_TI_SCI_FW_RGN_ID.

There is no functional change for non am62x flavors, just added separate entry for SA2UL_TI_SCI_FW_RGN_ID.
though there is some duplication since most devices are the same, but it looks cleaner and more consistent.

Had discussion with Andrew Davis internally on this, he suggested the same. 

Please let me know your thoughts.

Comment (jforissier) [reply core/arch/arm/plat-k3/platform_config.h:81]: Oh, I'm sorry, I missed the fact that you removed the common line:
```diff
- #define SA2UL_TI_SCI_FW_RGN_ID	0
```

So indeed all is fine. I will merge this. Thanks!
